gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/writer/HiveMetadataWriter.java

```diff
@@ -95,20 +95,21 @@ public HiveMetadataWriter(State state) throws IOException {
   @Override
   public void flush(String dbName, String tableName) throws IOException {
     String tableKey = tableNameJoiner.join(dbName, tableName);
-    log.info("start to flush table: " + tableKey);
-    HashMap<List<String>, ListenableFuture<Void>> executionMap =
-        this.currentExecutionMap.computeIfAbsent(tableKey, s -> new HashMap<>());
-    //iterator all execution to get the result to make sure they all succeeded
-    for (HashMap.Entry<List<String>, ListenableFuture<Void>> execution : executionMap.entrySet()) {
-      try {
-        execution.getValue().get(timeOutSeconds, TimeUnit.SECONDS);
-      } catch (InterruptedException | ExecutionException | TimeoutException e) {
-        log.error("Error when getting the result of registration for table" + tableKey);
-        throw new RuntimeException(e);
+    if(this.currentExecutionMap.containsKey(tableKey)) {
+      log.info("start to flush table: " + tableKey);
+      HashMap<List<String>, ListenableFuture<Void>> executionMap = this.currentExecutionMap.get(tableKey);
```

Comment:
just to my own understanding, what the key (typed as `List<String`) here?

Reply:
It's the partition value. Since we want to support multi partition here, it's a list. But most of the time, it only contains one element.